frameworks/helloworld/src/test/java/com/mesosphere/sdk/helloworld/scheduler/ServiceTest.java

```diff
@@ -550,18 +550,22 @@ public void expect(ClusterState state, SchedulerDriver mockDriver) throws Assert
                     .collect(Collectors.toMap(Phase::getName, p -> p));
             Assert.assertEquals(stepCounts.size(), phases.size());
             for (StepCount stepCount : stepCounts) {
-                Assert.assertEquals(getStepStatuses(state, stepCount),
+                Assert.assertEquals(getExpectedStepStatuses(state, stepCount),
                         phases.get(stepCount.phaseName).getChildren().stream()
                                 .collect(Collectors.toMap(Step::getName, Step::getStatus)));
             }
         }
 
-        private static Map<String, Status> getStepStatuses(ClusterState state, StepCount stepCount) {
+        private static Map<String, Status> getExpectedStepStatuses(ClusterState state, StepCount stepCount) {
             Map<String, Status> expectedSteps = new HashMap<>();
             expectedSteps.put(String.format("kill-%s-server", stepCount.phaseName),
                     stepCount.statusOfStepIndex(expectedSteps.size()));
-            for (String resourceId :
-                ResourceUtils.getResourceIds(ResourceUtils.getAllResources(state.getLastLaunchedPod(stepCount.phaseName)))) {
+            LaunchedPod pod = state.getLastLaunchedPod(stepCount.phaseName);
+
+            Collection<String> resourceIds = new ArrayList<>();
+            resourceIds.addAll(ResourceUtils.getResourceIds(ResourceUtils.getAllResources(pod.getTasks())));
+            resourceIds.addAll(ResourceUtils.getResourceIds(pod.getExecutor().getResourcesList()));
```

Comment:
Does this mean that if we write a test that forces a custom executor, all we need to do is ensure that `pod.getExecutor()` returns the correct resources?

Reply:
As of the latest revision just now, to use a custom executor you'd call `ServiceTestRunner::setUseCustomExecutor()` before calling `run()`.